test/matchers/matchers.go

```diff
@@ -0,0 +1,76 @@
+package matchers
+
+import (
+	"github.com/kyma-project/eventing-manager/api/v1alpha1"
+	"github.com/onsi/gomega"
+	gomegatypes "github.com/onsi/gomega/types"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func HaveBackendTypeNats(tp v1alpha1.Backend) gomegatypes.GomegaMatcher {
+	return gomega.And(
+		gomega.WithTransform(
+			func(e *v1alpha1.Eventing) string {
+				return e.Spec.Backends.Type
+			}, gomega.Equal(tp.Type)),
```

Comment:
Because the function name says `HaveBackendTypeNats`, does it make sense to always check for the type "NATS" a as the required type here?